src/Task.cc

```diff
@@ -3781,6 +3848,16 @@ void Task::dup_from(Task *other) {
       create_mapping(this, remote_this, km);
       LOG(debug) << "Copying mapping into " << tid;
       if (!(km.flags() & MAP_SHARED)) {
+        // Make the effort just for bigger mappings, copy smaller as a whole.
+        if (km.flags() & (MAP_ANONYMOUS | MAP_NORESERVE) &&
+            km.size() >= 0x10000000/*256MB*/)
```

Comment:
We could probably shrink this threshold to say 4MB and I don't think we need to check for `NORESERVE`.